### PR TITLE
implemented authorization by cancancan gem.

### PR DESCRIPTION
**Pull request by ichiragrathod**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -75,4 +75,5 @@ end
 gem 'devise'
 gem 'file_validators'
 gem 'rolify'
-gem 'rubocop', require: false
+gem 'cancancan'gem 'rubocop', require: false
+
```
Comment (poonamsimform) [inline Gemfile]: remove this extra line

File: Gemfile [hunk 2]
```diff
@@ -75,4 +75,5 @@ end
 gem 'devise'
 gem 'file_validators'
 gem 'rolify'
-gem 'rubocop', require: false
```
Comment (poonamsimform) [inline Gemfile]: Why rubocop removed?

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<br>
+<%=  link_to 'Posts listing', posts_path%>
+<br>
+<div class="card" style="width:400px">
+  <h5 class="card-header"><%= @post.title%></h5>
+  <div class="card-body">
+    <p class="card-text"><%= @post.body%></p>
+    <hr>
+    <% if can? :edit, @post %>
+      <%= link_to 'Edit', edit_post_path(@post.id), class:"btn btn-primary"%>
```
Comment (poonamsimform) [inline app/views/posts/show.html.erb]: check without passing id. How it make difference?

File: app/views/posts/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<h1>Update Post</h1>
+<%= render 'form'%>
+<%= link_to 'Cancel and Return to posts listing', posts_path%>
```
Comment (poonamsimform) [inline app/views/posts/edit.html.erb]: Please add this in main file so that no need to write the same line multiple times in multiple files

**Review by poonam-loconav — APPROVED** [review]

**Pull request merged**